File: server/src/main/java/com/example/server/note/controller/MarkdownController.java
package com.example.server.note.controller;

import com.example.server.note.service.MarkdownServiceImpl;
import com.example.server.note.domain.Markdown;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Optional;

@RestController
@RequiredArgsConstructor
public class MarkdownController {
    private final MarkdownServiceImpl markdownService;

    @GetMapping("/notes/{slug}/markdown")
    public Optional<Markdown> getMarkdown(@PathVariable String slug) {
        return markdownService.getMarkdown(slug);
    }

    @PatchMapping("/notes/{slug}/markdown")
    public Markdown patchMarkdown(@PathVariable String slug, @RequestBody Markdown md) {
        return markdownService.updateMarkdown(slug, md.getContent());
    }
}

File: server/api-server/src/main/kotlin/work/notespace/server/note/controller/MarkdownController.kt
package work.notespace.server.note.controller

import jakarta.validation.Valid
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*
import work.notespace.server.note.controller.request.MarkdownRequest
import work.notespace.server.note.dto.MarkdownDto
import work.notespace.server.note.service.MarkdownService

@RestController
@Validated
class MarkdownController(
    private val markdownService: MarkdownService
) {

    @GetMapping("/notes/{slug}/markdown")
    fun getMarkdown(
        @PathVariable
        @Valid
        slug: String
    ): MarkdownDto {
        return markdownService.getMarkdown(slug)
    }

    @PatchMapping("/notes/{slug}/markdown")
    fun patchMarkdown(
        @PathVariable
        slug: String,
        @RequestBody
        @Valid
        markdown: MarkdownRequest
    ): MarkdownDto {
        return markdownService.updateMarkdown(slug, markdown.content)
    }
}
